File: Backend/barbershop/barbershop/Models/ResponseDTOs/AppointmentDTO.cs
using barbershop.Models.Entitys;
using System;

namespace barbershop.Models.ResponseDTOs
{
    public class AppointmentDTO
    {
        public long AppointmentId { get; set; }

        public long UserId { get; set; }

        public int BarberId { get; set; }

        public int BranchId { get; set; }

        public DateOnly? AppointmentDate { get; set; }

        public TimeOnly? AppointmentTime { get; set; }

        public string Status { get; set; } = null!;

        public DateTime? CreatedAt { get; set; }

        public string? Note { get; set; }

        public bool? IsActive { get; set; }

        public List<AppointmentServiceDTO>? AppointmentServiceDTO { get; set; } = new List<AppointmentServiceDTO>();

        public  EmployeeDTO? EmployeeDTO { get; set; } = null!;

        public  BranchDTO? BranchDTO { get; set; } = null!;

        public List<PaymentDTO>? PaymentDTO { get; set; } = new List<PaymentDTO>();

        public List<ReviewDTO>? ReviewDTO { get; set; } = new List<ReviewDTO>();

        public UserDTO? UserDTO { get; set; } = null!;

        //Thêm 
        public List<ServiceDTO>? ServiceDTOs { get; set; }
    }
}

File: Frontend/barbershop/src/types/ResponseDTOs/appointmentDTO.ts
import type { AppointmentServiceDTO } from './appointmentServiceDTO';
import type { EmployeeDTO } from './employeeDTO';
import type { BranchDTO } from './branchDTO';
import type { PaymentDTO } from './paymentDTO';
import type { ReviewDTO } from './reviewDTO';
import type { UserDTO } from './userDTO';
import type { ServiceDTO } from './serviceDTO';
import type { BarberDTO } from './barberDTO';
export type AppointmentDTO = {
    appointmentId?: number;
    userId: number;
    barberId: number;
    branchId: number;
    appointmentDate?: string; // ISO date string
    appointmentTime?: string; // ISO time string
    status: string;
    createdAt?: string; // ISO date-time string
    note?: string;
    isActive?: boolean;
    appointmentServiceDTO?: AppointmentServiceDTO[];
    employeeDTO?: EmployeeDTO;
    barberDTO?: BarberDTO;
    branchDTO?: BranchDTO;
    paymentDTO?: PaymentDTO[];
    reviewDTO?: ReviewDTO[];
    userDTO?: UserDTO;
    serviceDTOs?: ServiceDTO[];
};
